pool.go

```diff
@@ -28,26 +31,70 @@ const (
 	tierMainToUnknown  = "main-to-unknown"
 	tierUnknownToMain  = "unknown-to-main"
 	BackendOverrideKey = "CABOOSE_BACKEND_OVERRIDE"
+	CabooseJwtIssuer   = "caboose-client"
 )
 
+// authenticateReq adds authentication to a request when a JWT_SECRET is present as an environment variable.
+func authenticateReq(req *http.Request, key string) (*http.Request, error) {
```

Comment:
@AmeanAsad We need a unit test for this function too.  Give this function a test http req object and ensure it comes back with what we expect it to.